tests/TestModified.cpp

```diff
@@ -417,53 +418,167 @@ void TestModified::testHistoryItem()
     QCOMPARE(entry2->historyItems().size(), 2);
 
     entry2->beginUpdate();
-    entry2->attachments()->remove("test");
+    entry2->attachments()->remove(key);
     entry2->endUpdate();
-    QCOMPARE(entry2->historyItems().size(), 0);
+    QCOMPARE(entry2->attachments()->attachmentsSize(), 0);
+    QCOMPARE(entry2->historyItems().size(), 1);
```

Comment:
Why this change for the historyItems size?

Reply:
Because the previous attachment gets copied into the history exceeding the configured maximum size.

Reply:
Mmmh ok seems reasonable